packages/vite/src/node/optimizer/index.ts

```diff
@@ -633,19 +633,33 @@ function parseOptimizedDepsMetadata(
   depsCacheDir: string,
   processing: Promise<DepOptimizationResult | undefined>
 ) {
-  const metadata = JSON.parse(jsonMetadata)
+  const metadata = JSON.parse(jsonMetadata, (key: string, value: string) => {
+    // Paths can be absolute or relative to the deps cache dir where
+    // the _metadata.json is located
+    if (key === 'file' || key === 'src') {
+      return normalizePath(path.resolve(depsCacheDir, value))
```

Comment:
I wonder if it is safe to use `normalizePath` since on Windows it would be converted to `/` instead of `\`, which wouldn't point to an actual filesystem file. Perhaps we don't need the `normalizePath` here?

Reply:
IIUC, this was already the case before the PR in Windows. Im triyng not to change that. But the path should still work with resolve, no? And since they are relative they could be copied across systems?

Reply:
Hmm if currently it is already normalised like `C:/foo/bar` then it should be fine then. I haven't confirmed this but looks a bit odd though, since it should be backslashes. My main concern is that during the initial optimisation run, the paths in memory won't be the same as the one in the next server startup (which loads from the cache). It could be working for now but would be a risk for bugs in the future.

Reply:
Paths in memory are also normalized

Reply:
Ah ok. Thanks for clarifying that. This should be good then 👍